File: src/main/java/com/gitlab/daring/image/command/parameter/EnumParam.java
package com.gitlab.daring.image.command.parameter;

import static com.gitlab.daring.image.util.EnumUtils.findEnum;

public class EnumParam<T extends Enum<T>>  extends CommandParam<T> {

	public Class<T> enumClass;

	public EnumParam(Class<T> cl, String sv) {
		super(sv, "");
		enumClass = cl;
		v = parseValue(args[0]);
	}

	public int vi() {
		return v.ordinal();
	}

	@Override
	public T parseValue(String sv) {
		return enumClass != null ? findEnum(enumClass, sv) : null;
	}

	@Override
	public Object configValue() {
		return "" + v;
	}
}

File: src/main/kotlin/com/gitlab/daring/image/command/parameter/EnumParam.kt
package com.gitlab.daring.image.command.parameter

import com.gitlab.daring.image.util.EnumUtils.findEnum

class EnumParam<T : Enum<T>>(val enumClass: Class<T>, sv: String) : CommandParam<T>(sv, "") {

    val enumValues get() = enumClass.enumConstants

    val vi get() = v.ordinal

    override fun parseValue(sv: String) = findEnum(enumClass, sv)

    override fun configValue(): Any  = "$v"

}